experiments/testers/ReadMe.md

```diff
@@ -0,0 +1,27 @@
+
+However, if the testing tools evaluate a given algorithm using an epsilon that is smaller than the epsilon used to generate noise, then a differential privacy violation occurs. This is because `insufficient_noise` has been applied to the algorithm; in other words, the algorithm is claiming more privacy protection than it actually provides.
+
+For all algorithms with insufficient_noise, the advertised_epsilon, or the epsilon used to evaluate the algorithm for differential privacy, is always less than the implemented_epsilon, or the actual amount of noise applied to the algorithm. This deliberate mismatch ensures that the algorithm will always violate differential privacy, because it is always claiming to have more privacy protection than the true amount of noise applied to the data.
+
+The relationship between the advertised_epsilon and the implemented_epsilon can be represented by a third variable, ratio, where ratio = advertised_epsilon / implemented epsilon. For example, if advertised_epsilon = 2 and implemented_epsilon = 5, then ratio = 0.4, meaning that only 40% of privacy protection that is claimed is actually being applied.
+
+## Architecture
+
+The evaluation involves running both C++ and Java files. This is because the purpose of the evaluation is to compare the Stochastic Tester, which is written in C++, to the Statistical Tester, which is written in Java.
+
+Under typical circumstances, the Stochastic Tester evaluates differentially private algorithms constructed from the C++ library, while the Statistical Tester evaluates differentially private algorithms constructed from the Java library. However, to maintain a consistent environment for testing, both tools in this evaluation rely upon the differential privacy algorithms from the C++ library. For the Java-based Statistical Tester, the evaluation constructs the algorithms ahead of time and generates samples using the C++ library, then passes them into the Statistical Tester for assessment. 
+
+The evaluation contains two folders: `cc` and `java`. The `cc` folder must be run first. The C++ files construct algorithms and generate samples from the C++ library and locate them in the `java` folder. Each set of samples has its own folder based on algorithm type (e.g., `countsamples`). Then the Stochastic Tester is run over a series of algorithms with `insufficient_noise`, and results are recorded in the `results` folder.
+
```

Comment:
Somehow, it's not sufficiently clear/obvious that data samples are generated from the C++ library, and then used by the Java library. I realise you do kind of state it a number of times in this section, but somehow, in the way the section is worded and organized, I don't think it's clear enough.

Also, it would be good to explain *why* we do it this way.